### DISCUSSION TITLE
Is there a any restriction on the number of properties that we can have for a egde?

### DISCUSSION THREAD

**Discussion by susanth19** [vesoft-inc/nebula#4465] [Q&A]
Hi All,

Is there any restriction on the number of properties that we can have for an edge? such that it doesn't affect the performance of our data model and at the same time performance of query traversing also.

Thanks

**Comment by wey-gu** [comment]
Hi @susanth19 

I had a similar question answered here: https://github.com/vesoft-inc/nebula/discussions/4364#discussioncomment-3215482

And I am quoting it here:

Fetching edge properties underlying is a key seek in RocksDB, the srcid, dstid, edge_type, rank will be the key of the record and property in blob will be the value.

When the property is large(that is we define a huge amount of property columns per edge type), the performance in pure FETCH could be impacted from the caching perspective, that is, fewer data could be placed in memory. To mitigate that, Key value separation could be leveraged.

ref: https://docs.nebula-graph.io/3.2.0/5.configurations-and-logs/1.configurations/4.storage-config/#key-value_separation_configurations

note, we should at best reduce property when possible during the processing of data before putting them into graph databases, but when it has to be put into properties fields(with a bunch of them, try leveraging k-v separation :).
